src/Treasure/Tui/GroupBorder.hs

```diff
@@ -0,0 +1,87 @@
+{- |
+@Brick@ library helper functions to group checkbox elements inside the form.
+
+This is not going to be the part of the library itself, so we will have it in
+our own libraries. See relevant discussion under the corresponding issue:
+
+* https://github.com/jtdaugherty/brick/issues/190
+
+-}
+
+module Treasure.Tui.GroupBorder
+       ( groupBorder
+       ) where
+
+import Brick (Edges (..), Padding (Max), Widget, padRight, str, vLimit, (<+>), (<=>))
+import Brick.Forms (FormFieldState, (@@=))
+
+import qualified Brick.Widgets.Border as B
+import qualified Brick.Widgets.Border.Style as B
+import qualified Brick.Widgets.Core as B
+
+
+{- |
+This function unites any amount of the form elements under the one group in
+bourders with the given group name. Intended to be used for joining check-boxes
+together, but any other elemens of the form will work the same way.
+
+__Example:__
+
+@
+┌─────────────────Form───────────────────┐
+│                                        │
+│┌──────────────Accounts────────────────┐│
+││[ ] user1                             ││
+││[ ] User2                             ││
+│└──────────────────────────────────────┘│
+└────────────────────────────────────────┘
+@
+
+**Note:** on an empty list it doesn't create any group or border.
+-}
+groupBorder :: String -> [s -> FormFieldState s e n] -> [s -> FormFieldState s e n]
+groupBorder groupName  = \case
+    []       -> []
+    [x]      -> [groupAllBorders groupName x]
+    (x:y:xs) -> let (mid, l) = (init $ y :| xs, last $ y :| xs) in
```

Comment:
This actually proves one thing that people don't need unsafe `init` and `last` that work with ordinary list. Usually you can prove to compiler that you're working with non-empty structure :+1: